compiler/src/iree/compiler/Codegen/Common/FoldTensorSubsetIntoVectorTransferOps.cpp

```diff
@@ -117,6 +119,47 @@ class FoldExtractSliceIntoTransferRead final
   }
 };
 
+/// Returns true if `writeOp` fully overwrites its destination.
+static bool isDestinationFullyOverwritten(vector::TransferWriteOp writeOp) {
+  if (writeOp.hasOutOfBoundsDim())
+    return false;
+  if (writeOp.getVectorType().getRank() != writeOp.getShapedType().getRank())
+    return false;
+  if (writeOp.getMask())
+    return false;
+
+  std::optional<iree_compiler::VscaleRange> vscaleRange;
+  auto vecType = writeOp.getVectorType();
+  if (vecType.isScalable()) {
+    auto targetAttr =
+        iree_compiler::IREE::HAL::ExecutableTargetAttr::lookup(writeOp);
+    vscaleRange = iree_compiler::getDefaultVscaleRange(targetAttr);
+  }
+
+  auto dest = writeOp.getSource();
+  auto destShape = writeOp.getShapedType().getShape();
+  auto destDimSize =
+      [&](unsigned dimIndex) -> FailureOr<iree_compiler::DimBoundSize> {
+    auto size = destShape[dimIndex];
+    if (size != ShapedType::kDynamic)
+      return iree_compiler::DimBoundSize{size};
+    return iree_compiler::computeDimUpperBound(dest, dimIndex, vscaleRange);
+  };
```

Comment:
This is a basically a function - please follow the naming convention for functions:
* https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly

Unless there's some other guideline for lambdas?

Also, could you document what this lambda does? Specifically for those of us unfamiliar with `computeDimUpperBound`?